log4j-kit/src/main/java/org/apache/logging/log4j/kit/env/support/PropertySourcePropertyEnvironment.java

```diff
@@ -0,0 +1,94 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to you under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.logging.log4j.kit.env.support;
+
+import java.util.Collection;
+import java.util.Comparator;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.concurrent.ConcurrentHashMap;
+import org.apache.logging.log4j.Logger;
+import org.apache.logging.log4j.kit.env.ConfigurablePropertyEnvironment;
+import org.apache.logging.log4j.kit.env.PropertyEnvironment;
+import org.apache.logging.log4j.kit.env.PropertySource;
+import org.apache.logging.log4j.kit.env.internal.CopyOnWriteNavigableSet;
+import org.jspecify.annotations.Nullable;
+
+public class PropertySourcePropertyEnvironment extends ClassloaderPropertyEnvironment
```

Comment:
Piotr, `PropertySourcePropertyEnvironment`... just reading the class name is making my mind whirl. I get totally confused of what is a `PropertySource`, what was a `PropertyEnvironment`, etc. In essence, this is a `ClassLoaderPropertyEnvironment` with some sources. Can't we say `SourcedPropertyEnvironment` instead, please?

Reply:
I renamed it to `CompositePropertyEnvironment`.